utils.py

```diff
@@ -6673,8 +6673,42 @@ def isGoodForNERSC(self, no_step=False):
         #good = good &
         return good
 
+    def getTimeInfoForChain(self):
+        timeInfo = dict()
+        if 'Chain' in self.request['RequestType']:
+            base = self.request['RequestType'].replace('Chain','')
+            itask=1
+            carry_on = {}
+            while True:
+                t = '%s%d'%(base, itask)
+                itask+=1
+                if t in self.request:
+                    task = self.request[t]
+                    timeInfo[t] = {
+                        'tpe':task.get('TimePerEvent')/task.get('FilterEfficiency',1),
+                        'cores':task.get('Multicore',1)
```

Comment:
Dima, `Multicore` is an optional argument and it might be present only at the top level (even though it could be that McM always set them in each task). So you might want to change this line from
```
'cores':task.get('Multicore',1)
```
to
```
'cores':task.get('Multicore', self.request['Multicore'])
```

Reply:
done